api/src/routes/certificate.test.ts

```diff
@@ -0,0 +1,435 @@
+            is2018DataVisCert: false,
+            isApisMicroservicesCert: false,
+            isInfosecQaCert: false,
+            isQaCertV7: false,
+            isInfosecCertV7: false,
+            isFrontEndCert: false,
+            isBackEndCert: false,
+            isDataVisCert: false,
+            isFullStackCert: false,
+            isSciCompPyCertV7: false,
+            isDataAnalysisPyCertV7: false,
+            isMachineLearningPyCertV7: false,
+            isRelationalDatabaseCertV8: false,
+            isCollegeAlgebraPyCertV8: false,
+            isFoundationalCSharpCertV8: false
+          },
+          completedChallenges: [
+            {
+              completedDate: 123456789,
+              files: [],
+              id: 'bd7158d8c442eddfaeb5bd18'
+            },
+            {
+              completedDate: 123456789,
+              files: [],
+              id: '587d78af367417b2b2512b03'
+            },
+            {
+              completedDate: 123456789,
+              files: [],
+              id: '587d78af367417b2b2512b04'
+            },
+            {
+              completedDate: 123456789,
+              files: [],
+              id: '587d78b0367417b2b2512b05'
+            },
+            {
+              completedDate: 123456789,
+              files: [],
+              id: 'bd7158d8c242eddfaeb5bd13'
+            },
+            {
+              challengeType: 7,
+              // TODO: use matcher for date near now
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+              completedDate: expect.any(Number),
+              files: [],
+              id: '561add10cb82ac38a17513bc'
+            }
+          ]
+        });
+        expect(response.status).toBe(200);
+      });
+
+      // Tests for all certifications as to what may currently be claimed, and what may no longer be claimed
+      test('should return 400 if certSlug is not allowed', async () => {
+        const claimableCerts = [
+          Certification.RespWebDesign,
+          Certification.JsAlgoDataStruct,
+          Certification.FrontEndDevLibs,
+          Certification.DataVis,
+          Certification.RelationalDb,
+          Certification.BackEndDevApis,
+          Certification.QualityAssurance,
+          Certification.SciCompPy,
+          Certification.DataAnalysisPy,
+          Certification.InfoSec,
+          Certification.MachineLearningPy,
+          Certification.CollegeAlgebraPy,
+          Certification.FoundationalCSharp,
+          Certification.LegacyFrontEnd,
+          Certification.LegacyBackEnd,
+          Certification.LegacyDataVis,
+          Certification.LegacyInfoSecQa,
+          Certification.LegacyFullStack
+        ];
+        const unclaimableCerts = ['fake-slug'];
+
+        if (SHOW_UPCOMING_CHANGES) {
+          claimableCerts.push(Certification.UpcomingPython);
+        } else {
+          unclaimableCerts.push(Certification.UpcomingPython);
+        }
```

Comment:
Conditional testing is a bit odd, tbh. I'd just test both or one of them (without the conditional).

Please merge regardless, I just wanted to make the observation!